pettingzoo/magent/magent_env.py

```diff
@@ -118,6 +113,7 @@ def reset(self):
         self.env.reset()
         self.frames = 0
         self.all_dones = {agent: False for agent in self.possible_agents}
+        self.team_sizes = [self.env.get_num(handle) for handle in self.handles]
```

Comment:
It could be cleaner to refactor and call this reset() function in __init__. I was trying it out but it started requiring more changes than I expected, leading to other things irrelevant to this PR.